### ISSUE TITLE
Docker file permission issue after upgrading PCF

### ISSUE THREAD

**Issue by cah-andrewfitzgerald** [cloudfoundry/garden-runc-release#79]
# Description

We recently upgraded to PCF 1.12.19, and a (prometheus) docker image that used to work is now failing with file permission issues 

# Environment

* garden-runc-release version: 1.12.1
* IaaS: AWS
* Stemcell version: light-bosh-stemcell-3468.30-aws-xen-hvm-ubuntu-trusty-go_agent
* Kernel version: not sure, can't start container

# Steps to reproduce

`cf push some-app -o prom/prometheus`

# Logs

```
   2018-05-15T09:07:09.16-0400 [CELL/0] OUT Creating container
   2018-05-15T09:07:09.39-0400 [CELL/0] OUT Successfully destroyed container
   2018-05-15T09:07:09.76-0400 [CELL/0] OUT Successfully created container
   2018-05-15T09:07:10.01-0400 [CELL/0] OUT Starting health monitoring of container
   2018-05-15T09:07:10.18-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.179828917Z caller=main.go:220 msg="Starting Prometheus" version="(version=2.2.1, branch=HEAD, revision=bc6058c81272a8d938c05e75607371284236aadc)"
   2018-05-15T09:07:10.18-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.179928593Z caller=main.go:221 build_context="(go=go1.10, user=root@149e5b3f0829, date=20180314-14:15:45)"
   2018-05-15T09:07:10.18-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.179978934Z caller=main.go:222 host_details="(Linux 4.4.0-119-generic #143~14.04.1-Ubuntu SMP Mon Apr 2 18:04:36 UTC 2018 x86_64 b939328f-96c4-4131-61b3-d721 (none))"
   2018-05-15T09:07:10.18-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.180063828Z caller=main.go:223 fd_limits="(soft=16384, hard=16384)"
   2018-05-15T09:07:10.18-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.189856754Z caller=main.go:504 msg="Starting TSDB ..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.19017001Z caller=web.go:382 component=web msg="Start listening for connections" address=0.0.0.0:9090
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190173907Z caller=main.go:398 msg="Stopping scrape discovery manager..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190233993Z caller=main.go:411 msg="Stopping notify discovery manager..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190271771Z caller=main.go:432 msg="Stopping scrape manager..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190361913Z caller=manager.go:460 component="rule manager" msg="Stopping rule manager..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190478723Z caller=manager.go:466 component="rule manager" msg="Rule manager stopped"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190605644Z caller=notifier.go:512 component=notifier msg="Stopping notification manager..."
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190643211Z caller=main.go:407 msg="Notify discovery manager stopped"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.190861107Z caller=main.go:394 msg="Scrape discovery manager stopped"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.191290319Z caller=main.go:426 msg="Scrape manager stopped"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.191310256Z caller=main.go:573 msg="Notifier manager stopped"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=error ts=2018-05-15T13:07:10.194150223Z caller=main.go:582 err="Opening storage failed open DB in /prometheus: open /prometheus/518918001: permission denied"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=info ts=2018-05-15T13:07:10.194206722Z caller=main.go:584 msg="See you next time!"
   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] OUT Exit status 0
   2018-05-15T09:07:10.19-0400 [CELL/0] OUT Exit status 0
   2018-05-15T09:07:10.21-0400 [CELL/0] OUT Stopping instance b939328f-96c4-4131-61b3-d721
   2018-05-15T09:07:10.21-0400 [CELL/0] OUT Destroying container
   2018-05-15T09:07:10.23-0400 [API/1] OUT Process has crashed with type: "web"
```
`   2018-05-15T09:07:10.19-0400 [APP/PROC/WEB/0] ERR level=error ts=2018-05-15T13:07:10.194150223Z caller=main.go:582 err="Opening storage failed open DB in /prometheus: open /prometheus/518918001: permission denied"`
# Docker image (if relevant)

prom/prometheus

# Cause

unsure, this worked before the PCF upgrade and fails afterwards

# Resolution

???

**Comment by Callisto13** [comment]
Hi @cah-andrewfitzgerald, thanks for submitting this issue.
Could you confirm that this is the Dockerfile used to build the image: https://hub.docker.com/r/prom/prometheus/~/dockerfile/
Could you also let me know which version you upgraded _from_.
Thanks!

**Comment by cah-andrewfitzgerald** [comment]
Confirming that is the correct Dockerfile.

@cah-jeremykuhn, do you happen to know the previous version?

**Comment by Callisto13** [comment]
Hey @cah-andrewfitzgerald. We have found that this failure was caused by the combination of [this security fix](https://github.com/prometheus/prometheus/issues/2939#issuecomment-314998558)
in the image, and the way that our rootfs provider (GrootFS) handled volumes in Garden-Runc-Release 1.12.1.
In versions 1.13.1 (PCF 1.12.22) and later, GrootFS will lay down volumes as the `nobody` user, so your docker image should work.
GRR 1.13.1 does have some other known issues, but volumes is not one of them :) .
